backend/director/tools/serp.py

```
@@ -0,0 +1,73 @@
+
+        if duration:
+            if duration not in duration_mapping:
+                raise ValueError(f"Invalid duration value: {duration}")
+            params["tbs"] = duration_mapping[duration]
+
+        try:
+            response = self.session.get(self.base_url, params=params, timeout=self.timeout)
+            response.raise_for_status()
+            return response.json().get("video_results", [])
+        except requests.exceptions.RequestException as e:
+            raise RuntimeError(f"Error during SerpAPI video search: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)